File: src/main/java/agh/ics/oop/model/MapDirection.java
package agh.ics.oop.model;

public enum MapDirection {
    NORTH,
    EAST,
    SOUTH,
    WEST;

    public String toString() {
        return switch (this) {
            case NORTH -> "Północ";
            case SOUTH -> "Południe";
            case WEST -> "Zachód";
            case EAST -> "Wschód";
        };
    }

    public MapDirection next() {
        int nextOrdinal = (this.ordinal() + 1) % MapDirection.values().length;
        return MapDirection.values()[nextOrdinal];
    }

    public MapDirection previous() {
        int nextOrdinal = (this.ordinal() + 3) % MapDirection.values().length;
        return MapDirection.values()[nextOrdinal];
    }

    public Vector2d toUnitVector() {
        return switch (this) {
            case NORTH -> new Vector2d(0, 1);
            case SOUTH -> new Vector2d(0, -1);
            case WEST -> new Vector2d(-1, 0);
            case EAST -> new Vector2d(1, 0);
        };
    }
}

File: kotlin/src/main/kotlin/model/MapDirection.kt
package model;

enum class MapDirection {
    NORTH,
    SOUTH,
    WEST,
    EAST;
    override fun toString() = when(this) {
        NORTH -> ("N")
        SOUTH -> ("S")
        WEST -> ("W")
        EAST -> ("E")
    }
    fun next() = when(this) {
        NORTH -> EAST
        SOUTH -> WEST
        WEST -> NORTH
        EAST -> SOUTH
    }
    fun previous() = when(this) {
        NORTH -> WEST
        WEST -> SOUTH
        SOUTH -> EAST
        EAST -> NORTH
    }
}